src/main/java/com/atlassian/db/replica/api/aurora/AuroraCluster.java

```diff
@@ -0,0 +1,70 @@
+package com.atlassian.db.replica.api.aurora;
+
+import java.util.Objects;
+
+public class AuroraCluster {
+    private final String clusterName;
+    private final String clusterPrefix;
+
+    public AuroraCluster(String clusterName) {
+        this(clusterName, null);
+    }
+
+    public AuroraCluster(String clusterName, String clusterPrefix) {
+        this.clusterName = clusterName;
+        this.clusterPrefix = clusterPrefix;
+    }
+
+    public static AuroraCluster fromString(String cluster) {
+        String[] chunks = splitByLastOccurence("-", cluster);
+        if (chunks.length == 1) {
```

Comment:
No need to worry about cluster==null?

Reply:
Not sure, everything can be null in Java :/

Reply:
Not after a null check 😉

Reply:
What is the win? We get exception anyway

Reply:
Let's fail fast, maybe it will be more predictable but still not type safe :/